internal/cmd/operator-sdk/cli/cli.go

```diff
@@ -21,57 +21,86 @@ import (
 	cfgv2 "sigs.k8s.io/kubebuilder/v3/pkg/config/v2"
 	cfgv3 "sigs.k8s.io/kubebuilder/v3/pkg/config/v3"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+	golangv2 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/v2"
+	golangv3 "sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/v3"
 
-	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/alpha"
 	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/alpha/config3alphato3"
 	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/bundle"
 	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/cleanup"
-	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/completion"
 	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/generate"
 	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/olm"
 	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/run"
 	"github.com/operator-framework/operator-sdk/internal/cmd/operator-sdk/scorecard"
 	"github.com/operator-framework/operator-sdk/internal/flags"
+	"github.com/operator-framework/operator-sdk/internal/plugins"
 	ansiblev1 "github.com/operator-framework/operator-sdk/internal/plugins/ansible/v1"
-	golangv2 "github.com/operator-framework/operator-sdk/internal/plugins/golang/v2"
-	golangv3 "github.com/operator-framework/operator-sdk/internal/plugins/golang/v3"
+	envtestv1 "github.com/operator-framework/operator-sdk/internal/plugins/envtest/v1"
 	helmv1 "github.com/operator-framework/operator-sdk/internal/plugins/helm/v1"
+	manifestsv2 "github.com/operator-framework/operator-sdk/internal/plugins/manifests/v2"
+	scorecardv2 "github.com/operator-framework/operator-sdk/internal/plugins/scorecard/v2"
 	"github.com/operator-framework/operator-sdk/internal/util/projutil"
 )
 
-var commands = []*cobra.Command{
-	alpha.NewCmd(),
-	bundle.NewCmd(),
-	cleanup.NewCmd(),
-	completion.NewCmd(),
-	generate.NewCmd(),
-	olm.NewCmd(),
-	run.NewCmd(),
-	scorecard.NewCmd(),
-}
+var (
+	commands = []*cobra.Command{
+		bundle.NewCmd(),
+		cleanup.NewCmd(),
+		generate.NewCmd(),
+		olm.NewCmd(),
+		run.NewCmd(),
+		scorecard.NewCmd(),
+	}
+	alphaCommands = []*cobra.Command{
+		config3alphato3.NewCmd(),
+	}
+)
 
 func Run() error {
-	cli, _ := GetPluginsCLIAndRoot()
-	return cli.Run()
+	c, _ := GetPluginsCLIAndRoot()
+	return c.Run()
 }
 
 // GetPluginsCLIAndRoot returns the plugins based CLI configured to use operator-sdk as the root command
 // This CLI can run kubebuilder commands and certain SDK specific commands that are aligned for
 // the kubebuilder project layout
-func GetPluginsCLIAndRoot() (cli.CLI, *cobra.Command) {
+func GetPluginsCLIAndRoot() (*cli.CLI, *cobra.Command) {
+	ansibleBundle, _ := plugin.NewBundle("ansible"+plugins.DefaultNameQualifier, plugin.Version{Number: 1},
+		ansiblev1.Plugin{},
+		manifestsv2.Plugin{},
+		scorecardv2.Plugin{},
+	)
+	gov2Bundle, _ := plugin.NewBundle(golangv2.Plugin{}.Name(), golangv2.Plugin{}.Version(),
```

Comment:
Overriding a plugin key feels like a bad idea. If I have a kubebuilder project that I want to use with operator-sdk, the CLI could fail because the `operator-sdk` binary thinks PROJECT says it uses the bundle when it actually uses the plugin.

Reply:
Additionally, operator-sdk can no longer register kubebuilder's go plugin directly.

Reply:
This alignment achieves the exact same behavior as right now. Right now, by using the plugin "go.kubebuilder.io/v2" or "go.kubebuilder.io/v3" you run the other plugins too. We are doing the same, which is kind of hacky as you mention. Now it is a bit more clear that we are doing it, but it is exactly the same.

As of the fails you mention, a project built with `kubebuilder init` will still work if using `operato-sdk create api` and the extra plugins will not be run (the only extra plugin that implements something outside of init is `manifests/v2`, which checks for the existence of its jey in the config, which is only present if the init was done by `operator-sdk init`). So, it will work, and no extra plugins will be used.

Regarding registering kubebuilder's go plugin, there is no real restriction. I do agree that it would be better to use the operator-sdk domain for the go bundles, therefore making a distinction between the k9r plugin and SDK's bundle, but that would break backwards compatibility.

Reply:
HI @estroz, 

See that it keeps the same current behaviour and does not change anything for the end-users. We cannot change the name of the plugins as discussed before. See: